crypto/asn1/tasn_utl.c

```diff
@@ -78,30 +78,34 @@ int ossl_asn1_do_lock(ASN1_VALUE **pval, int op, const ASN1_ITEM *it)
     aux = it->funcs;
     if (aux == NULL || (aux->flags & ASN1_AFLG_REFCOUNT) == 0)
         return 0;
-    lck = offset2ptr(*pval, aux->ref_offset);
     lock = offset2ptr(*pval, aux->ref_lock);
+    refcnt = offset2ptr(*pval, aux->ref_offset);
 
     switch (op) {
     case 0:
-        *lck = ret = 1;
+        if (!CRYPTO_NEW_REF(refcnt, 1))
+            return 0;
         *lock = CRYPTO_THREAD_lock_new();
```

Comment:
Is this actually necessary any more? This lock is not used by us anywhere AFAICT. Strictly speaking the "ref_lock" field in the ASN1_AUX structure is public...but are we required to actually initialise it to anything, or can we just set it to NULL?

Reply:
Yes...in theory...

Reply:
We've maintained things in the name of backward compatibility that were less justified IMO.

Leaving it as NULL would be okay if the locking functions returned when passed a NULL argument.  They don't, they crash.